edx_rest_framework_extensions/auth/jwt/decoder.py

```diff
@@ -65,17 +66,17 @@ def jwt_decode_handler(token):
         InvalidTokenError: Decoding fails.
     """
     jwt_issuer = get_first_jwt_issuer()
-    _verify_jwt_signature(token, jwt_issuer)
+    _verify_jwt_signature(token, jwt_issuer, decode_symmetric_token=decode_symmetric_token)
     decoded_token = _decode_and_verify_token(token, jwt_issuer)
     return _set_token_defaults(decoded_token)
 
 
-def configured_jwt_decode_handler(token):
+def configured_jwt_decode_handler(token, decode_symmetric_token=True):
```

Comment:
Since `configured_jwt_decode_handler` has a setting and expected signature with one argument, token, I recommend we leave this function as-is.

Instead, let's add a new function:
```
get asymmetric_only_jwt_decode_handler(token):
   """
   Returns a jwt_decode_handler that will only validate asymmetrically signed JWTs.

   WARNING: This will only work with a service that is configured to use the
       jwt_decode_handler from this library. This can be used to decode an
       already decoded JWT, to ensure it is asymmetrically signed. This check
       can go away once the DEPR for symmetrically signed JWTs is complete:
       https://github.com/openedx/public-engineering/issues/83
   """
```
This call can be used on the `request.auth` validated JWT token from LMS.